### PR TITLE
Allow for tutorials to be represented as not taken place

### LINKED ISSUE #81: Represent tutorials that have not occurred
Tutorials shouldn't just be marked red if not attended as it isn't entirely accurate as the tutorial might not have occurred yet, hence tutorials should maintain a third state if not being occurred yet.

### PR DESCRIPTION
**Pull request by sayglenn**
This PR aims to create a third state for tutorials, which is to not have taken place. Resolves #81.

Significant changes made:
- Created an `AttendanceStatus` enum to properly represent tutorial status.
- Changed the logic of tutorials in person from `Set<Tutorial>` to `Map<Tutorial, AttendanceStatus>`.
- Changed the storage method for tutorials to store both a number and attendance status.
- Amended the GUI to allow representation of three states of tutorial. Added grey to represent tutorials that have not taken place. 
- Fix the tutorial count to be 12, since it is the typical tutorial count for a semester.

### CODE REVIEW

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 1]
```diff
@@ -59,12 +62,32 @@ public static Set<Tag> getTagSet(String... strings) {
     }
 
     /**
-     * Returns a tutorial set containing the list of strings given.
+     * Returns a tutorial set containing the list of strings given and all are defaulted to true.
```
Comment (raageth) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:65]: Should be defaulted to not taken place yet.

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 2]
```diff
@@ -61,13 +62,18 @@ public PersonCard(Person person, int displayedIndex) {
                 .sorted(Comparator.comparing(tag -> tag.tagName))
                 .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
 
-        IntStream.rangeClosed(1, 10).forEach(index -> {
+        IntStream.rangeClosed(1, Person.MAXIMUM_TUTORIALS).forEach(index -> {
             Label tutorialLabel = new Label(String.valueOf(index));
             tutorialLabel.getStyleClass().add("tutorial-label");
 
-            if (person.hasAttendedTutorial(String.valueOf(index))) {
+            if (person.getAttendanceStatus(String.valueOf(index)) == AttendanceStatus.ATTENDED) {
```
Comment (raageth) [inline src/main/java/seedu/address/ui/PersonCard.java:69]: Should we have a method in person to do this check instead?

**Review by raageth — COMMENTED** [review]
Minor suggestions to improve overall code quality

File: src/main/java/seedu/address/ui/PersonCard.java (hunk 2 above)
Comment (sayglenn) [reply src/main/java/seedu/address/ui/PersonCard.java]: Lifted the method up to the Person class to retrieve CSS class.

File: src/main/java/seedu/address/model/util/SampleDataUtil.java (hunk 1 above)
Comment (sayglenn) [reply src/main/java/seedu/address/model/util/SampleDataUtil.java]: Thanks for catching this! Amended.

File: src/main/java/seedu/address/model/person/AttendanceStatus.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package seedu.address.model.person;
+
+/**
+ * Represents the Attendance Status of a tutorial. Mainly used for GUI generation and testing.
+ */
+public enum AttendanceStatus {
+    ATTENDED,
```
[the remark this answers is not in the document]
Comment (sayglenn) [reply src/main/java/seedu/address/model/person/AttendanceStatus.java]: Good suggestion, have amended it to align.

**Review by raageth — APPROVED** [review]

**Pull request merged**